src/mito2/src/sst/index/fulltext_index/applier.rs

```diff
@@ -110,14 +146,32 @@ impl FulltextIndexApplier {
         &self,
         file_id: FileId,
         column_id: ColumnId,
+        file_size_hint: Option<u64>,
     ) -> Result<Option<SstPuffinDir>> {
-        let puffin_manager = self.puffin_manager_factory.build(self.store.clone());
-        let file_path = location::index_file_path(&self.region_dir, file_id);
+        let mut puffin_manager = None;
+        if let Some(file_cache) = &self.file_cache {
+            let index_key = IndexKey::new(self.region_id, file_id, FileType::Puffin);
+            if file_cache.get(index_key).await.is_some() {
+                puffin_manager = Some(self.puffin_manager_factory.build(
+                    file_cache.local_store(),
+                    WriteCachePathProvider::new(self.region_id, file_cache.clone()),
+                ));
+            }
```

Comment:
The cache may remove the file before we get the dir. We may need to fallback to remote reader if we can't get the dir.